src/main/java/org/opentripplanner/graph_builder/module/osm/WayPropertySet.java

```diff
@@ -340,22 +374,35 @@ public void setProperties(String spec, StreetTraversalPermission permission) {
   public void setProperties(
     String spec,
```

Comment:
I think a more developer-friendly way of setting these values would be to pass in a (builder) object that makes the parameters clear:

```java
wps.setProperties(
  WayProperties.of("highway=trunk").permission(ALL).bicycleSafety(0.5, 0.5).walkSafety(0.8,0.8).mixin()
)
```
This is of course a lot more verbose but it tells you quite clearly what the various floats mean.

WFYT?

Reply:
Yes we sort of discussed this topic in the developer meeting as well. I do like the idea and I think I'll implement it today.

Reply:
So should this be implemented here or later on in a separate method? I think the verbosity can be lessened by a bit for most cases where we don't need the separate back safety.
```
wps.setProperties(
  WayProperties.of("highway=trunk").permission(ALL).bicycleSafety(0.5).walkSafety(0.8).mixin()
)
```

Reply:
Good idea.

I would be fine either way but personally I would do it in this PR.

I thought a little bit more about the semantics and I think I like this version best:

```java
// creating a mixin - can you override the traversal permission with a mixin?
wps.setMixinProperties(WayProperties.of("highway=trunk").bicycleSafety(0.5).walkSafety(0.8));

wps.setProperties(WayProperties.of("highway=trunk").permission(ALL).bicycleSafety(0.5).walkSafety(0.8));

```

Reply:
I can implemented this in this pr today.

Reply:
Implemented something similar in https://github.com/opentripplanner/OpenTripPlanner/pull/4328/commits/a160869d80e356b6a146743e49930ea5d5ec0189. I created a builder class and made the WayProperties immutable. The WayProperties class doesn't currently contain the OSM specification but not sure if it should.